python/CWE-079/XssFlaskAudit.ql

```diff
@@ -0,0 +1,49 @@
+/**
+ * @name Audit: Cross-site scripting using Flask Jinja2 templates
+ * @description Templates are vulnerable to cross-site scripting if they are
+ *              rendered with untrusted data.
+ * @kind path-problem
+ * @problem.severity warning
+ * @security-severity 2.5
+ * @sub-severity low
+ * @precision very-low
+ * @id py/audit/xss-jinja2
+ * @tags security
+ *       external/cwe/cwe-079
+ *       audit
+ */
+
+import python
+import DataFlow::PathGraph
+import semmle.python.Concepts
+import semmle.python.ApiGraphs
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.RemoteFlowSources
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.security.dataflow.ReflectedXSSCustomizations
+
+class DynamicTemplate extends DataFlow::Node {
```

Comment:
Isn’t there a way to do that using Data Extensions now? You just need to define flask.rendertemplate as an XSS sink using the Python data extensions - right?